pkg/cmd/util/clientcmd/factory.go

```diff
@@ -169,6 +171,39 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			return nil, fmt.Errorf("it is not possible to get ports from %s", kind)
 		}
 	}
+	w.LogsForObject = func(object runtime.Object, container string, follow, previous bool) (io.ReadCloser, error) {
+		client, err := clients.ClientForVersion("")
+		if err != nil {
+			return nil, err
+		}
+		switch t := object.(type) {
+		case *api.Pod:
+			if len(container) == 0 {
+				if len(t.Spec.Containers) != 1 {
+					return nil, fmt.Errorf("POD %s has more than one container; please specify the container to print logs for with -c", t.ObjectMeta.Name)
+				}
+				container = t.Spec.Containers[0].Name
+			}
+			return client.RESTClient.Get().Namespace(t.Namespace).Name(t.Name).
```

Comment:
I thought we had access to a full `client.Interface` to make these calls.  We should use a `BuildLogsInterface`, right?